src/ir.cc

```diff
@@ -651,4 +651,52 @@ Const::Const(F64Tag, uint64_t value, const Location& loc_)
 Const::Const(V128Tag, v128 value, const Location& loc_)
     : loc(loc_), type(Type::V128), vec128(value) {}
 
+uint8_t ElemSegment::GetFlags(const Module* module) const {
+  uint8_t flags = 0;
+
+  switch (kind) {
+    case SegmentKind::Active: {
+      Index table_index = module->GetTableIndex(table_var);
+      if (table_index != 0) {
```

Comment:
Doesn't this mean that specifying table_index of 0 explicitly doesn't round trip?

Reply:
Yes, but I think that's OK. We don't guarantee a perfect roundtrip (one example is if you use over-long LEB encoding). The semantics are otherwise unchanged.